pyam/core.py

```diff
@@ -437,7 +437,7 @@ def variables(self, include_units=False):
         )
 
     def append(self, other, ignore_meta_conflict=False, inplace=False,
-               **kwargs):
+               verify_integrity=True, **kwargs):
```

Comment:
I would prefer to not (even optionally) allow the possibility to have duplicate rows  in the timeseries data...

Reply:
`is_unique` on a long multiindex is actually quite an expensive operation. i would like to give the knowledgable user a safety belt off option where he is responsible for ensuring integrity or where he is responsible for checking integrity.

The main use case in my mind was a small improvement for pyam.concat where we call append with `verify_integrity = False` and then call `verify_index_integrity` once the final dataframe has been constructed. (Converting that function directly to `pd.concat` as proposed in #501 is way better though).

Reply:
Fair enough... Doesn't stand in the way of doing both this improvement and improving `concat()` in the separate PR